SwiftUIDemoAppUITests/Tests/DeeplinkTests.swift

```diff
@@ -0,0 +1,147 @@
+        let content = try XCTUnwrap(String(data: contentData, encoding: .utf8))
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .replacingOccurrences(of: "\n", with: "")
+            .replacingOccurrences(of: " ", with: "")
+
+        XCTAssertEqual(content, """
+        {"applinks":{"apps":[],"details":[{"appID":"EHV7XZLAHA.io.getstream.iOS.VideoDemoApp","paths":["/video/demos/*","/video/demos"]}]}}
+        """)
+    }
+
+    func test_universalLink_production_joinsExpectedCall() {
+        WHEN("") {
+            Safari.openUniversalLinkFromSmartBanner(MockDeeplink.production)
+        }
+        THEN("user joins the the specified call") {
+            userRobot
+                .assertCallControls()
+                .assertParticipantsAreVisible(count: 1)
+        }
+    }
+
+    func test_customSchemeURL_joinsExpectedCall() {
+        WHEN("User opens a URL that contains a custom scheme") {
+            Safari.openApp(MockDeeplink.customScheme)
+        }
+        THEN("user joins the the specified call") {
+            userRobot
+                .assertCallControls()
+                .assertParticipantsAreVisible(count: 1)
+        }
+    }
+}
+
+extension XCUIDevice {
+
+    fileprivate func open(_ url: URL) {
+        DispatchQueue.main.async {
+            // Need to async this because XCUIDevice.shared.system.open
+            // synchronously waits for a button to be pressed the first time.
+            self.system.open(url)
+        }
+    }
+}
+
+enum Safari {
+
+    private static let safari = XCUIApplication(bundleIdentifier: "com.apple.mobilesafari")
+
+    fileprivate static func go(to url: URL) {
+        safari.textFields["Address"].tap()
+        safari.typeText(url.absoluteString)
+        let goButton = safari.buttons["Go"]
+        if goButton.waitForExistence(timeout: 5) {
+            safari.tapKeyboardKey("Go")
+        }
+    }
+
+    fileprivate static func open(_ url: URL) {
+        safari.launch()
+
+        _ = safari.wait(for: .runningForeground, timeout: 5)
+
+        // Type the deeplink and execute it
+        let firstLaunchContinueButton = safari.buttons["Continue"]
+        if firstLaunchContinueButton.exists {
+            firstLaunchContinueButton.tap()
+        }
+
+        go(to: url)
+    }
+
+    fileprivate static func openApp(_ url: URL) {
+        open(url)
+
+        let openButton = safari.buttons["Open"]
+        if openButton.waitForExistence(timeout: 5) {
+            openButton.tap()
+        }
+    }
+
+    fileprivate static func openUniversalLinkFromSmartBanner(_ url: URL) {
+        open(url)
+
+        let allowButton1 = safari.alerts.buttons["Allow"]
+        if allowButton1.waitForExistence(timeout: 5) {
+            allowButton1.tap()
+            let allowButton2 = safari.alerts.buttons["Allow"]
+            if allowButton2.waitForExistence(timeout: 5) {
+                allowButton2.tap()
+            }
```

Comment:
`safari.alerts.buttons["Allow"]` is just a locator, it does not change, I think there is no need to assign it multiple times 🤔

```suggestion
        let allowButton = safari.alerts.buttons["Allow"]
        if allowButton.waitForExistence(timeout: 5) {
            allowButton.tap()
            if allowButton.waitForExistence(timeout: 5) {
                allowButton.tap()
            }
```


By the way, if these `allow` buttons appear all the time, we can use `wait()`, something like this:

```suggestion
        let allowButton = safari.alerts.buttons["Allow"]
       allowButton.wait().tap()
       allowButton.wait().tap()
```